torch/__init__.py

```diff
@@ -398,55 +456,55 @@ def __rpow__(self, other):
             return sym_float(self).__rpow__(sym_float(other))
 
     def __eq__(self, other: object) -> builtins.bool:
-        raise AssertionError("type stub not overridden")
+        raise NotImplementedError("type stub not overridden")
```

Comment:
AssertionError is right here. We have some initialization code elsewhere which is supposed to override all of these. If there is a problem with that code, this is a framework bug and we should assert fail

Reply:
NotImplementedError implies that someday someone in the framework will implement it. That's 100% not the intent here, if you hit this code, it is a BUG. These stubs exist purely to improve type checking.

Reply:
> NotImplementedError implies that someday someone in the framework will implement it.

I see. Seems that `RuntimeError` or `TypeError` would be better.